util/chunk/chunk.go

```diff
@@ -556,6 +606,27 @@ func (c *Chunk) Column(colIdx int) *Column {
 	return c.columns[colIdx]
 }
 
+// Sel returns Sel of this Chunk.
+func (c *Chunk) Sel() []int16 {
+	return c.sel
+}
+
+// SetSel sets a Sel for this Chunk.
+func (c *Chunk) SetSel(sel []int16) {
+	c.sel = sel
+}
+
+// Reconstruct removes all unselected rows according to Sel.
+func (c *Chunk) Reconstruct() {
```

Comment:
Do we need to reconstruct self?
Maybe copy to another chunk is enough?

Reply:
It seems you are right.
I remove it now and add it back when we actually need it.